sources/System/Bonus/PlayerBonusSystem.cpp

```diff
@@ -17,9 +21,20 @@ namespace BBM
 		auto &playerBonus = entity.get<PlayerBonusComponent>();
 
 		playerBonus.nextNoClipRate -= dtime;
-		if (playerBonus.nextNoClipRate <= 0ns) {
+		if (playerBonus.nextNoClipRate <= 0ns && playerBonus.isNoClipOn) {
 			playerBonus.nextNoClipRate = playerBonus.noClipBonusRate;
 			playerBonus.isNoClipOn = false;
+			auto playerPos = entity->tryGetComponent<PositionComponent>();
+			auto playerHealth = entity->tryGetComponent<HealthComponent>();
+			if (!playerHealth || !playerPos)
+				return;
+			for (auto &[other, pos, _] : this->_wal.getScene()->view<PositionComponent, CollisionComponent>()) {
```

Comment:
You should probably iterate over boxes or something. If you do it like that, the player will be killed if the no-clip ends while he is on a bonus.

Reply:
True